library/proc_macro/src/bridge/closure.rs

```diff
@@ -1,27 +1,26 @@
 //! Closure type (equivalent to `&mut dyn FnMut(A) -> R`) that's `repr(C)`.
 
-use std::marker::PhantomData;
-
 #[repr(C)]
-pub struct Closure<'a, A, R> {
+pub struct Closure<A, R> {
```

Comment:
This lifetime parameter is important as `Env` only lives for this amount of time. `Closure` is an FFI safe wrapper around `&mut FnMut`, not around `Box<FnOnce>`.

Reply:
Yeah, it's only coincidental that the same `PhantomData` is used for both `'a` and `Send`/`Sync` - you would simply not have the lifetime in there if all you wanted was to remove `Send`/`Sync`.

See the module level comment comparing this type to `&mut dyn FnMut(A) -> R`.

Reply:
I will add a comment explaining this additional info.